include/systems/diff_system.h

```diff
@@ -123,7 +123,8 @@ class DifferentiableSystem : public ImplicitSystem,
    */
   virtual void assembly (bool get_residual,
                          bool get_jacobian,
-                         bool apply_heterogeneous_constraints = false) libmesh_override = 0;
+                         bool apply_heterogeneous_constraints = false,
+			 bool apply_no_constraints = false) libmesh_override = 0;
```

Comment:
Aha, add
```
(setq-default indent-tabs-mode nil)
```
to your .emacs file, as we don't allow mixes of tabs and spaces for indentation.